RealJenkinsRuleInit/src/main/java/org/jvnet/hudson/test/RealJenkinsRuleInit.java

```diff
@@ -36,6 +44,30 @@ public RealJenkinsRuleInit() {}
 
     @Override
     public void start() throws Exception {
+        /*
+        According to ServletRequest documentation:
+            If the parameter data was sent in the request body, such as occurs with an HTTP POST request,
+            then reading the body directly via getInputStream or getReader can interfere with the execution of this method.
+        Check RealJenkinsRule#runRemotely, it has
+         conn.setRequestProperty("Content-Type", "text/plain; utf-8");
+
+         to avoid this issue.
+        */
+        PluginServletFilter.addFilter(new Filter() {
```

Comment:
I am not sure what this is doing here. Is this part of the patch fixing some bug, or demonstrating it?

Reply:
This is a filter I've added to make the tests fail. 
With the ContentType to text it works.

Reply:
Well revert this and do it properly in a single test case.

Reply:
I think adding complexity to the current plugin should be enough. By adding complexity to the current plugin used more cases are covered.
If I had a new test case for this particular case only, the options of having an issue when a filter is used are bigger.

Reply:
Not following what you are saying. To repeat: if the bug is fixed by the one-line change to `RealJenkinsRule.java`, then please revert all changes to this file, and add a test case to `RealJenkinsRuleTest.java` which reproduces the error when you

```bash
git checkout master -- src/main
```

Reply:
`RealJenkinsRule.java` only has one line change. Regarding `RealJenkinsRuleTest` I think I needs no change.
In order to reproduce the bug I had to change `RealJenkinsRuleInit` to add a filter to the plugin. 
I do not know how to add a filter to a plugin from the test file.
Another option would be to add another plugin `RealJenkinsRuleInitWithFilter`.  Do you want that?

Reply:
> I do not know how to add a filter to a plugin from the test file.

Just this same code, but from a `runRemotely` block. Presumably you will need to then run another no-op `runRemotely` block to reproduce the error after adding the filter.

Reply:
Thanks. Should be able to do a final review Monday.

Reply:
> which reproduces the error

Checked locally. Does fix the error, though I do not get any colorful details, just

```
java.io.IOException: Server returned HTTP response code: 500 for URL: http://localhost:63078/jenkins/RealJenkinsRule/step?token=8b4e3d47-75e9-4cba-a236-51e6514d381c
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1900)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1498)
	at org.jvnet.hudson.test.RealJenkinsRule.runRemotely(RealJenkinsRule.java:449)
	at org.jvnet.hudson.test.RealJenkinsRuleTest.testFilter(RealJenkinsRuleTest.java:62)
```